### ISSUE TITLE
Coq version?

### ISSUE THREAD

**Issue by Blaisorblade** [bobatkey/system-f-parametricity-model#1]
Hi @bobatkey! I thought this could be helpful, but which Coq version was this developed with? Unsurprisingly, It doesn't compile out-of-the-box with Coq 8.4pl1 because it's too new. Me trying to quick-fix the code sounds like a bad idea :-| (I tried starting and I was already unsure of the first fix — I'd need to see the old code before doing that).

**Comment by Blaisorblade** [comment]
More specifically, when I tried this with Coq 8.4 I ended up changing the implementation of `subst`.
With the quickfix below, I have no idea whether the implementation is the right one, and I can't check that without taking a look at the original Coq's output.

```
diff --git a/TypeSyntax.v b/TypeSyntax.v
index e7929e1..ad5d235 100644
--- a/TypeSyntax.v
+++ b/TypeSyntax.v
@@ -66,8 +66,8 @@ intros. dependent induction H0.
  apply subst_var with (d:=c + 1 + g). assumption. reflexivity. assumption.
  (* ty_arr *)
  apply ty_arr.
-  apply IHtype1. apply H. reflexivity.
-  apply IHtype2. apply H. reflexivity.
+  apply IHtype1. apply H. (* reflexivity. *)
+  apply IHtype2. apply H. (* reflexivity. *)
  (* ty_forall *)
  apply ty_forall.
  apply IHtype with (c0:=S c). apply shift1. apply H. reflexivity.
```

**Comment by bobatkey** [comment]
Hello, thanks for reporting this. I've fixed this now. Your fix was correct. I've also fixed the other places where Coq 8.4pl6 complains. They were all due to changes in how the `dependent induction` tactic works. As far as I can see, this tactic has become a bit better at reducing the amount of manual proving of equalities that you have to do, hence the removal of `reflexivity` in `TypeSyntax.v` and other places.

**Comment by Blaisorblade** [comment]
Well, thanks a lot for the fix!
